src/gateway/fetch.ts

```diff
@@ -66,7 +66,7 @@ export class Fetch extends Gateway {
     }
 
     const headers = assign(customHeaders, this.request.headers())
-    const method = this.shouldEmulateHTTP() ? 'post' : requestMethod
+    const method = this.shouldEmulateHTTP() ? 'POST' : requestMethod.toUpperCase()
```

Comment:
The same should be done in two more files, I know this was originally only for Fetch gateway but the way I see it this is not about your specific case and changing Fetch to be different just because of undici:
- `src/gateway/http.ts`
- `src/gateway/xhr.ts`